mempool/src/pool.rs

```diff
@@ -852,17 +816,29 @@ where
                     "collect_txs: next tx has ancestor score {:?}",
                     next_tx.ancestor_score()
                 );
-                tx_accumulator.add_tx(next_tx.tx().clone());
+
+                match tx_accumulator.add_tx(next_tx.tx().clone(), next_tx.fee()) {
+                    Ok(_) => (),
+                    Err(err) => log::error!(
+                        "CRITICAL: Failed to add transaction {} from mempool. Error: {}",
```

Comment:
Shouldn't we break out of this loop if we fail to add a transaction? It seems to me like this will run forever.

Reply:
It won't run forever, because the iterator will keep proceeding. It'll keep trying again and again though, until the transaction times out in the mempool and gets dropped. But then we should notice this in the logs. This function doesn't have the authority to remove transactions from the mempool.